### ISSUE TITLE
Style/IdenticalConditionalBranches misses empty when branches

### ISSUE THREAD

**Issue by pt-stripe** [bbatsov/rubocop#4143]
## Expected behavior

No error

## Actual behavior

```
$ cat a.rb
case foo
when 1
  3
when 2
else
  3
end
$ rubocop a.rb --only Style/IdenticalConditionalBranches
Inspecting 1 file
C

Offenses:

a.rb:3:3: C: Move 3 out of the conditional.
  3
  ^
a.rb:6:3: C: Move 3 out of the conditional.
  3
  ^

1 file inspected, 2 offenses detected
```

## RuboCop version

Include the output of `rubocop -V`. Here's an example:

```
$ rubocop -V
0.47.1 (using Parser 2.4.0.0, running on ruby 2.2.5 x86_64-darwin16)
```

**Comment by dpostorivo** [comment]
While the message is incorrect, the intent of the cop seems to be correct.  Why would you do: 

```
case foo
when 1
  3
when 2
else
  3
end
```

When you could do:
```
case foo
when 2
else
  3
end
```
 or even in this case

```
3 unless foo == 2
```
I started to work on a patch for this, but changing the message for this case seems more appropriate.

**Comment by pocke** [comment]
@dpostorivo Your example is special case. If the bodies do not match perfectly, we can't do the refactoring.

For example:

```ruby
case foo
when 1
  do_something
  3
when 2
else
  do_something2
  3
end
```

**Comment by dpostorivo** [comment]
@pocke  so new cop for this special case when the bodies mach perfectly? I did just finish a patch for this, but I felt the case in and of itself should be caught by rubocop based on the OP's case so I didn't want to submit a pull request.

**Comment by pt-stripe** [comment]
I clearly gave you a simplified example. We ran into this in practice in a few places in our large codebase.

It is a bug with the cop, since in this case it (correctly) doesn't suggest moving it out:

```
case foo
when 1
  3
when 2
  nil # This line is different from the original
else
  3
end
```
